### ISSUE TITLE
Indexing arrays programatically with a variable

### ISSUE THREAD

**Issue by yousifBilal** [ics-jku/wal#31]
Hello,

we have a case where we want to access an array element programatically:
```
(set [some_index 3])
(reval my_signal.my_array<some_index> 0)
```
This however does not work. It is reading the whole my_signal.my_array<some_index> as a variable name and I can't seem to figure out how to make it evaluate some_index so that eventually compiles to this: my_signal.my_array<3>.

I managed to do it by defining a custom operator:
```python
index_signal = lambda seval, args: seval.eval(
    str(args[0]) + "<" + str(seval.eval(args[1])) + ">"
)
wal.register_operator("index_signal", index_signal)
```
then doing 
```
(define indexed_value (index_signal "my_signal.my_array" some_index))
```
Only then I can successfully evaluate the index.

It would be nice if whatever between <> is evaluated rather then being taken as is.

**Comment by yousifBilal** [comment]
Thank you for the reply.

I see. I guess I wasn't aware of the symbol-add operator. So this is what we use to create a symbol that we have to build at runtime, am I correct?

**Comment by LucasKl** [comment]
Yes, with symbol add you can concatenate multiple symbols, strings, or numbers at runtime. E.g.
```lisp
>-> (symbol-add 'a 'b 3 (+ 2 2) "-def")
ab34-def
```

**Comment by yousifBilal** [comment]
Great! Maybe consider adding it to the documentation so other people can see the feature. You can close the issue.
